### PR TITLE
Updated for PWA 6

### PR DESCRIPTION
**Pull request by dqdquint**
## Description

Migrated boilerplate extension to PWA 6.

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md (did not do this because it is a template)

### CODE REVIEW

File: frontend/actions/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {
 /**
  * Get product swatches action.
  * @param {string} dummyId dummyId
- * @returns {function}
+ * @returns {func}
```
Comment (Carsten89) [inline frontend/actions/index.js]: why did you change it from function to func ?

File: frontend/components/DummyPortal/connector.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const mapStateToProps = state => ({
 
 /**
  * Maps dispatch to props.
- * @param {function} dispatch Dispatch.
+ * @param {func} dispatch Dispatch.
```
Comment (Carsten89) [inline frontend/components/DummyPortal/connector.js]: why did you change it from function to func ?

File: frontend/subscriptions/index.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { main$ } from '@shopgate/pwa-common/streams/main';
 
 /**
  * Subscriptions
- * @param {function} subscribe Subscribe.
+ * @param {func} subscribe Subscribe.
```
Comment (Carsten89) [inline frontend/subscriptions/index.js]: why did you change it from function to func ?

File: extension-config.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <% if (frontend.enabled) { %>
     {
       "id": "Dummy",
-      "path": "frontend/routes/Dummy.jsx",
+      "path": "frontend/components/DummyRoute/index.jsx",
```
Comment (Carsten89) [inline extension-config.json]: for me it should still be in "pages" or "routes"

File: frontend/components/DummyRoute/index.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        </View>
+      )}
+    </Theme>
+  );
+}
+
+/**
+ * @returns {JSX}
+ */
+function DummyPageRoute() {
```
Comment (Carsten89) [inline frontend/components/DummyRoute/index.jsx]: i would prefer something like this:
`
export default () => (
 <Route pattern={DUMMY_PATH} component={DummyPage} />);`

File: frontend/package.json [hunk 6]
```diff
@@ -28,9 +31,12 @@
...
     "eslint": "^4.19.1",
+    "glamor": "^2.20.40",
     "prop-types": "^15.6.0",
     "react": "^16.2.0",
     "react-dom": "^16.2.0",
+    "react-helmet": "^5.2.0",
+    "react-transition-group": "^2.2.1",
     "react-hot-loader": "^3.1.3",
     "react-transform-catch-errors": "^1.0.2"
   }
```
Comment (Carsten89) [inline frontend/package.json:48]: please add peerDependencies here

**Review by Carsten89 — CHANGES_REQUESTED** [review]

File: frontend/actions/index.js (hunk 1 above)
Comment (dqdquint) [reply frontend/actions/index.js]: eslint wanted this updated

File: frontend/components/DummyPortal/connector.js (hunk 2 above)
Comment (dqdquint) [reply frontend/components/DummyPortal/connector.js]: eslint wanted this updated

File: frontend/subscriptions/index.js (hunk 3 above)
Comment (dqdquint) [reply frontend/subscriptions/index.js]: eslint wanted this updated

File: extension-config.json (hunk 4 above)
Comment (dqdquint) [reply extension-config.json]: Moved to routes folder

File: frontend/components/DummyRoute/index.jsx (hunk 5 above)
Comment (dqdquint) [reply frontend/components/DummyRoute/index.jsx]: Updated

File: frontend/package.json (hunk 6 above)
Comment (dqdquint) [reply frontend/package.json:48]: Added peerDependencies

File: frontend/package.json [hunk 7]
```diff
@@ -13,11 +13,14 @@
   },
   "devDependencies": {
     <% if (frontend.lint) { %>"@shopgate/eslint-config": "*",<% } %>
+    "@shopgate/eslint-config": "*",
```
Comment (Carsten89) [inline frontend/package.json]: this gets already added one line above

File: frontend/actions/index.js (hunk 1 above)
Comment (Carsten89) [reply frontend/actions/index.js]: i think you should check your eslint settings. 
"ESLint: Use 'Function' instead of 'function'. (valid-jsdoc)" this is the ESLint error i get

File: frontend/package.json (hunk 6 above)
Comment (Carsten89) [reply frontend/package.json:48]: i changed all our dependencies from "*" to "^6.3.1"

File: frontend/routes/DummyRoute/index.jsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import { Route } from '@shopgate/pwa-common/components';
+import { Theme } from '@shopgate/pwa-common/context';
+import { DUMMY_PATH } from '../../constants';
+
+/**
+ * @returns {JSX}
+ */
+function DummyPage() {
+  return (
+    <Theme>
```
Comment (Carsten89) [inline frontend/routes/DummyRoute/index.jsx]: i talked with szymon and we both think, we should use our extension kit here

**Review by Carsten89 — CHANGES_REQUESTED** [review]

File: frontend/actions/index.js (hunk 1 above)
Comment (dqdquint) [reply frontend/actions/index.js]: Doubled checked in PWA and confirmed use of Function. Updated

File: frontend/components/DummyPortal/connector.js (hunk 2 above)
Comment (dqdquint) [reply frontend/components/DummyPortal/connector.js]: Used Function instead

File: frontend/subscriptions/index.js (hunk 3 above)
Comment (dqdquint) [reply frontend/subscriptions/index.js]: Used Function instead

File: frontend/package.json (hunk 6 above)
Comment (dqdquint) [reply frontend/package.json:48]: Added   "dependencies": { "@shopgate-ps/pwa-extension-kit": "^0.6.0"  },

File: frontend/routes/DummyRoute/index.jsx (hunk 8 above)
Comment (dqdquint) [reply frontend/routes/DummyRoute/index.jsx]: Updated to use extension kit

File: frontend/package.json (hunk 7 above)
Comment (dqdquint) [reply frontend/package.json]: ah right. Thanks for removal

File: frontend/routes/DummyRoute/index.jsx (hunk 8 above)
Comment (sznowicki) [reply frontend/routes/DummyRoute/index.jsx]: Well, not really:
https://github.com/shopgate-professional-services/pwa-extension-kit/blob/master/src/connectors/README.md#withThemeComponents

You're wrapping component with `withThemeComponents` but still you're using the Theme context.

**Review by sznowicki — CHANGES_REQUESTED** [review]
Please finish implementation of `withThemeComponents`

File: frontend/routes/DummyRoute/index.jsx (hunk 8 above)
Comment (dqdquint) [reply frontend/routes/DummyRoute/index.jsx]: I see. Updated!

**Comment by sznowicki** [comment]
I have updated this PR with some cleanups. I strongly believe that there's no need for connectors in both route and page. It's too much.

I moved connecting the page component into page component itself, so route stays simple.

**Review by sznowicki — APPROVED** [review]
